Formula/h/hctl.rb

```diff
@@ -0,0 +1,32 @@
+class Hctl < Formula
+  desc "Tool to control your Home Assistant devices from the command-line"
+  homepage "https://github.com/xx4h/hctl"
+  url "https://github.com/xx4h/hctl.git",
+      tag:      "v0.4.0",
+      revision: "e4fc037a284d28a7519ac1512d5201a2620ff623"
+  license "Apache-2.0"
+  head "https://github.com/xx4h/hctl.git", branch: "main"
+
+  livecheck do
+    url :stable
+    regex(/^v?(\d+(?:\.\d+)+)$/i)
+  end
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/xx4h/hctl/cmd.version=v#{version}
+      -X github.com/xx4h/hctl/cmd.commit=#{Utils.git_head}
+      -X github.com/xx4h/hctl/cmd.date=#{time.iso8601}
+    ]
+    system "go", "build", *std_go_args(ldflags:)
+
+    generate_completions_from_executable(bin/"hctl", "completion")
+  end
+
+  test do
+    assert_match "Hctl is a CLI tool to control your home automation", shell_output("#{bin}/hctl --help")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for your feedback!

It seemed like the tests failed anyway due to my repository being not yet "notable enough".

Anyway, i updated the test to verify that the tool failes with an expected error message when there is no config.